### ISSUE TITLE
About model robust

### ISSUE THREAD

**Issue by RaySue** [lukas-blecher/LaTeX-OCR#78]
Thx for your share. I trained a model follow your instruction, but it's not work on other image, only get valid result from trained image. Can you give some advice? Enlarge the model capacity or data augment ... any valid path, Thx.

**Comment by lukas-blecher** [comment]
Hi, can you tell me what performance (bleu, edit distance) your model achieves on the test set?
And can you share an example image where it isn't working?

Of course, you can use a bigger model. If there is enough data the performance should improve. Data augmentation is already in place 
https://github.com/lukas-blecher/LaTeX-OCR/blob/97c9644e776862234f71c0913b9b2a4e8da2fc15/dataset/dataset.py#L19-L36
But I'm open to suggestions.

**Comment by RaySue** [comment]
Thank you for you patient reply. 

Actually I trained model without your pretrain is not robust. yesterday, I trained model with your pretrain, the model is more robust than before, can work on test data , its achieve accept preformance.

---

**Training Data:**

I download hand writing formula data from the link you privided [here](https://drive.google.com/drive/folders/13CA4vAmOmD_I_dSbvLp-Lf0s6KiaNfuO) 

I split the data into two part, train 999  and test 822


**Padding Resize**
I only change the image loader，I resize all image into（128，1280）, specifically I resize the long side into 128 or 1280，and padding 255 for other side，final size is （128， 1280）to ensure the image is not distort.

**Experiment:**

I download the **weithts.pth** from your provided link, which is 98 M.
(all the experiment model is same to original config.yaml)

exp1:
Training model **without your pretrain** , batchsize=108, 
- 205 Epoch status:  
    - Avg Train loss 0.001, 
    - Test: BLEU: 0.27
    - The model on test data is not robust.

exp2:
Training model **with your pretrain** , batchsize=8,
- 90 Epoch status:
    - Avg Train loss 0.018
    - Test: BLEU 0.88
    - The model‘s perfomance on test data is in line with expection. 

--- 
My Question:

1. Why I trained model from scratch is not robust, the data is too less?
2. How did you train the pretrain model? Use how much data and which data?
3. Your original dataset.py is split image into differernt buckets by their shape, but I padding resize all images into unify size, is this will impact model convergence？

Thx

**Comment by lukas-blecher** [comment]
1. Yes, I believe the model is overfitting because there are too few training samples.
2. I used all the data from the training set. Why would you leave out most of it anyway?
3. I don't know. It should not be a huge problem, but why did you unify the size at all?
4. I'm not sure. I don't remember that part. I will have to investigate it.

Also, you can increase batch size if your machine supports it for exp2 :)

**Comment by RaySue** [comment]
2.Can you tell me all training set contain how many samples? Only hand writing or handwriting and print? 
3.I want to deploy the model for online service, I need to set cudnn.benchmark = true to reduce the latancy.

**Comment by lukas-blecher** [comment]
The training set from `formulae.zip` has ~160k samples. I did not try to train on hand written equations because I don't have the resources right now.

Can you confirm that the speedup of cudnn is greater than an adjustable image size? The model has less to do if the images are cropped to the right size. That's why I implemented it in that way.

As many samples as you can get :) It depends on the model size of course.
I think my pretrained model is also overfitting on the training set a little bit. More data is needed.
